language/diem-tools/writeset-transaction-generator/src/main.rs

```diff
@@ -47,6 +51,50 @@ enum Command {
         /// Blockchain height
         version: u64,
     },
+    /// Verify if a blob is generated by the checked-in artifact.
+    #[structopt(name = "verify-release")]
+    VerifyDiemFrameworkRelease {
+        /// Path to the serialized bytes of WriteSet.
+        #[structopt(parse(from_os_str))]
+        writeset_path: PathBuf,
+        /// True if the serialized byte has the type of WriteSetPayload.
+        #[structopt(long)]
+        is_payload: bool,
```

Comment:
This seems funky. It would probably be simpler not to allow polymorphism in the type of the blob. Is there a reason this tries to support validating both a WriteSet transaction and a payload?

Reply:
The main problem here is this: writeset used for AOS has different type from the writeset used for waypoint. That's why we need the output-payload flag and the is-payload flag here. Although this command is mostly used for the AOS for release process, in rare cases where we want to roll out emergency changes via waypoint this flag will be needed. E.g: if we find a critical secruity breach in Move modules on chain

Reply:
I replaced this flag with a either or logic with reasons given above: we want to be able to verify both types because in extreme cases we might want to create a release when the system is down, then we have to rely on the Waypoint Transaction to bring us back.

Reply:
Another question is should we distinguish the two types of payload by enforcing a extension name for a blob? That we would be able to get rid of the polymorphism over there.

Reply:
I like the extension-based approach.